codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyFrameCodec.java

```diff
@@ -125,13 +125,19 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
 
     @Override
     public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
-        if (!read) {
+        boolean wasRead = read;
+        read = false;
+
+        // Discard bytes of the cumulation buffer if needed.
+        discardSomeReadBytes();
+
+        if (!wasRead) {
             if (!ctx.channel().config().isAutoRead()) {
                 ctx.read();
             }
+        } else {
+            ctx.fireChannelReadComplete();
         }
-        read = false;
-        super.channelReadComplete(ctx);
```

Comment:
`ByteToMessageDecoder` has some state which is updated in its `channelReadComplete` ... shouldn't we still delegate?